### DISCUSSION TITLE
I don't know how to use conditional

### DISCUSSION THREAD

**Discussion by ler93** [jspsych/jsPsych#2068] [Q&A] [answered]
Hi! I find myself stuck in a problem that surely is easy to solve.

The situation is that I must create a stimulus that, depending on the participant's response, takes him to one page or another of the experiment. That is, if the subject answers "yes" to the question, then he should see a new question to answer on the screen. Instead, if his/her answer is "no", should skip that question and continue with the experiment.
The first step I made was to type the following:

```
var final_questions1 = {
        type: 'html-button-response',
        stimulus: 'While you were reading the phenomenon: Did you remember any of the stories read during the text comprehension activity?',
        choices: ['Yes', 'No'],
    };
    timeline.push (final_questions1);
```

After this the conditional parameter should go (I guess). If the answer to the last question was "yes", then they should answer this:

```
var final_questions2 = {
        type: 'survey-text',
        questions: [
          {prompt: '<p> Can you describe in as much detail as possible what was the situation you remembered? </p>',
          columns: 60, rows: 10, required: true, name: 'Memory of situation'}],
          button_label: 'Next>',
    };
    timeline.push (final_questions_conditional);
```

If instead the answer in "final_questions1" was "no", they should go to another survey-text that continue with the experiment.

I hope I was clear in the description.
From already thank you very much!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @ler93, you can have a look at the documentation for conditional timelines [here](https://www.jspsych.org/overview/timeline/#conditional-timelines). 

In your case, the `final_questions2` trial should go inside of a conditional node, since that trial should only run under certain conditions. In the `conditional_function` for your conditional node, you'll want to check what choice was made on the `final_questions1` trial. The conditional function should return `true` if "yes" was selected, and should return `false` if "no" was selected. 

There are a few different ways that you can access the data from the `final_questions1` trial. One option is to get all of the jsPsych data with [`jsPsych.data.get`](https://www.jspsych.org/core_library/jspsych-data/#jspsychdataget), and then filter the resulting data collection with the [`filter` function](https://www.jspsych.org/core_library/jspsych-data/#filter). But if `final_questions1` always comes immediately before the conditional node in your timeline, then it's probably easier to use the [`jsPsych.data.getLastTrialData` function](https://www.jspsych.org/core_library/jspsych-data/#jspsychdatagetlasttrialdata). 

Here's what this looks like all together:
```js
var final_questions1 = {
  type: 'html-button-response',
  stimulus: 'While you were reading the phenomenon: Did you remember any of the stories read during the text comprehension activity?',
  choices: ['Yes', 'No'],  // 'yes' is 0, 'no' is 1
};
timeline.push (final_questions1);

// create this trial but don't add it directly to your main timeline
var final_questions2 = {
  type: 'survey-text',
  questions: [
    {prompt: '<p> Can you describe in as much detail as possible what was the situation you remembered? </p>',
    columns: 60, rows: 10, required: true, name: 'Memory of situation'}],
  button_label: 'Next>',
};

// create a conditional node that contains the final_question2 trial and a conditional function,
// and add this to the main timeline
var final_questions_conditional = {
  timeline: [final_questions2],
  conditional_function() {
    // get the data from the last trial (final_questions1)
    var last_trial_data = jsPsych.data.getLastTrialData().values()[0];
    if (last_trial_data.response == 0) {
      // if the response was choice 0 ("yes"), then run the conditional timeline
      return true;
    } else {
      // otherwise skip the conditional timeline
      return false;
    }
  }
};
timeline.push (final_questions_conditional);
```

**Reply by ler93** [reply]
Hi, Becky. I copied the code that you gave me but an error occurs in one of the conditional node line: 
```javascript
    if (last_trial_data.response == 0) {
```
The error says: Uncaught TypeError: Cannot read property 'response' of undefined

I don't know what could be happening.

**Reply by becky-gilbert** [reply]
Oops, sorry! I forgot that you need to get the raw data from the data collection that is returned from the `jsPsych.data.getLastTrialData` function before you can index it and get the response. You can do this using the [`values` function](https://www.jspsych.org/core_library/jspsych-data/#values) for data collections. So the line that gets the last trial data inside the conditional function should look like this:
```js
var last_trial_data = jsPsych.data.getLastTrialData().values()[0];
```
I'll edit my response above to avoid confusion.

**Reply by ler93** [reply]
Thank you very much, Becky! It works perfectly. I'm starting to understand the mechanics behind some jspsych functions and that makes me happy. Thanks a lot for your time!

**Reply by becky-gilbert** [reply]
No problem! Glad it works now, and sorry for the error in my original response 😬
